internal/controller/ansibleRun/ansibleRun.go

```diff
@@ -228,16 +253,43 @@ type external struct {
 }
 
 func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
-	/*re, changes, _ := c.pbCmd.ParseResults(ctx, mg)
+	cr, ok := mg.(*v1alpha1.AnsibleRun)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotAnsibleRun)
+	}
+	switch c.runner.AnsibleRunPolicy.Name {
+	case "ObserveAndDelete", "":
+		if c.runner.AnsibleRunPolicy.Name == "" {
+			meta.SetPolicyRun(mg, "ObserveAndDelete")
```

Comment:
Where can I find the method definition for `SetPolicyRun`?

Reply:
@morningspace I am pointing to my fork https://github.com/multicloudlab/crossplane-provider-ansible/pull/57/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R112 
this commit => 93336f6af258

Reply:
ah, I see, but why we have to modify crossplane-runtime since this is a specific feature to ansible provider, while crossplane-runtime is something generic and shared across all crossplane providers. I thought that we just need to simply use the common methods that can handle k8s resource annotation manipulation and would be sufficient. Something similar to a forked version of k8s provider here: https://github.com/IBM/crossplane-provider-kubernetes/blob/master/internal/controller/object/object.go#L326

Reply:
I agree, the scope of this annotation is limited to Ansible Provider. I will fix this :)